### PR TITLE
Port CRSF RC driver to new Serial UART API

### PR DESCRIPTION
**Pull request by katzfey**
### Solved Problem
Moved the CRSF RC driver over to the new Serial UART API so that it can be used by Qurt platform. Also, there was a loop that wasn't being processed properly by the Qurt compiler so it was slightly restructured to make it compile cleanly.

### CODE REVIEW

File: platforms/common/Serial.cpp [hunk 1]
```diff
@@ -130,6 +135,24 @@ bool Serial::setFlowcontrol(FlowControl flowcontrol)
...
+bool Serial::setSingleWireMode()
+{
+	return _impl.setSingleWireMode();
+}
+
+bool Serial::getSwapRxTxMode() const
+{
+	return _impl.getSwapRxTxMode();
+}
+bool Serial::setSwapRxTxMode()
```
Comment (PetervdPerk-NXP) [inline platforms/common/Serial.cpp:151]: Can we also include the prototypes for inverted mode?
The idea is when introducing a new interface you try to accommodate for all uses and write these prototypes out.

Comment (katzfey) [reply platforms/common/Serial.cpp:151]: Sure, I just added them.

**Comment by katzfey** [comment]
@PetervdPerk-NXP , @dagar It would be much appreciated if this could get reviewed, approved, and merged by the end of this month (March 31). It would mean a lot to us at ModalAI, Inc. as this is the final PR to getting baseline support for VOXL2 boards in mainline.

**Comment by Seungwon-Yeo** [comment]
Dooray! 메일 발송 실패 안내



  
  
    
    
  
  
    
메일 발송
실패 안내
    
  
  
    
        
          
***@***.***) 
          님께
보낸
메일이
전송되지
못하였습니다.
        
        
          실패 사유를 확인해보세요.
        
    
  
  
    
      
        
          
            * 받는 사람 : 
***@***.***) 
          
          
            * 발송 시간 : 
2024-03-29T05:18:00
          
          
            * 메일 제목 : 
Re: [PX4/PX4-Autopilot] Port CRSF RC driver to new Serial UART API (PR #22917)
          
          
            
              
                * 실패 사유 : 
              
              
                
대외비/기밀 메일은 외부 계정으로 발송이 제한되어 일부 메일 발송 실패(수신자가 전달 설정한 경우 포함)
                
              
            
                      
          
        
      
    
  
  
    
        
          이 메일은 발신전용으로 회신되지 않습니다.
          더 궁금하신 사항은
          ***@***.***
          으로 문의해 주시기 바랍니다.
        
    
  
  
    
        © Dooray!.

**Review by dagar — APPROVED** [review]

**Review by dagar — APPROVED** [review]

**Pull request merged**